raymon/profiling/extractors/vision/anomaly.py

```diff
@@ -1,70 +1,84 @@
+        numpy_img = np.resize(numpy_img, (224, 224, 3))
+        numpy_img = numpy_img.reshape([1, 3, 224, 224]) 
+        return numpy_img
+    
+    def prepare_batch(self, images, batch_size):
+        # Our data is a list containing pil images
+        # Convert this image to numpy
+        # Make batches from data and store them in dataloader
+        dataloader = []
+        for i in range(0, len(images), batch_size):
+            # if we arrive the end of the list, we have to adjust the shape of the array,
+            if i + batch_size > len(images):
+                batch_numpy = np.zeros((len(images) - i, 3, 224, 224))
+            # Otherwise the shape of the array is (batch_size, 3, 224, 224).
+            else:
+                # Create zero numpy array with batch shape
+                batch_numpy = np.zeros((batch_size, 3, 224, 224))
+            # Make batches
+            for k in range(i, i + batch_size):
+                # Avoid to exceed the length of the images (list index out of range)
+                if k == len(images):
+                    break
+                x = 0 
+                # Convert pil image and do other preprocessings
+                numpy_img = self.preprocess(images[k])
+                batch_numpy[x, :, :, :] = numpy_img
+                x += 1 
+            dataloader.append(batch_numpy)
+        return dataloader
 
     def extract(self, data):
-        if not isinstance(data, Image.Image):
-            raise ValueError(f"data must be of type PIL.Image.Image, not {data.shape}")
-        batchtf = self.tfs(data)[None, :]
-        feats = self.mobilenet(batchtf).detach().numpy()
-        return super().extract(data=feats)
+        if not isinstance(data, Image.Image) and not isinstance(data, np.ndarray):
+            raise ValueError(f"type of data must be PIL.Image.Image or numpy.ndarray, not {type(data)}")
+        numpy_img = {self.mobilenet.get_inputs()[0].name: self.preprocess(data)}
+        feats = self.mobilenet.run(None, input_feed=numpy_img)
+        return super().extract(data=feats[0])
 
     def build(self, data, batch_size=16):
-        dataset = ImageDataset(loaded_data=data, transform=self.tfs)
-        # data is a list of images here
-        dataloader = DataLoader(dataset, shuffle=False, batch_size=batch_size, drop_last=True)
-        embeddings = torch.zeros(size=(len(data), 1000))
-        for batchidx, batchtf in enumerate(dataloader):
-            # batchtf = self.tfs(batch)
-            components = self.mobilenet(batchtf).detach()  # .numpy()
+        if not isinstance(data, Iterable):
+            raise ValueError(f"type of the data must be Iterable")
+        dataloader = self.prepare_batch(data, batch_size)               
+        embeddings = np.zeros((len(data), 1000))
+        for batchidx, batch_numpy in enumerate(dataloader):
+            batch_numpy = np.array(batch_numpy, dtype=np.float32)
+            ort_inputs2 = {self.mobilenet.get_inputs()[0].name: batch_numpy}
```

Comment:
`ort_inputs2` is a very cryptic variable name. And why is it '2'? Rename to something more sensible.